src/register/RegistrationPage.jsx

```diff
@@ -443,173 +436,123 @@ class RegistrationPage extends React.Component {
+                  platformName: this.state.platformName,
+                  tosAndHonorCode: (
+                    <Hyperlink destination={getConfig().TOS_AND_HONOR_CODE} target="_blank">
+                      {intl.formatMessage(messages['terms.of.service.and.honor.code'])}
+                    </Hyperlink>
+                  ),
+                  privacyPolicy: (
+                    <Hyperlink destination={getConfig().PRIVACY_POLICY} target="_blank">
+                      {intl.formatMessage(messages['privacy.policy'])}
+                    </Hyperlink>
+                  ),
+                }}
+              />
+            </div>
+            {this.state.optionalFields.length ? (
+              <Form.Group className="mb-0 mt-2 small">
+                <Form.Check
+                  id="optional-field-checkbox"
+                  type="checkbox"
+                  name="optionalFields"
+                  value={this.state.showOptionalField}
+                  onClick={this.handleOnChange}
+                  onChange={this.handleOnChange}
+                  label={intl.formatMessage(messages['support.education.research'])}
                 />
-                <div id="honor-code" className="pt-10 small">
-                  <FormattedMessage
-                    id="register.page.terms.of.service.and.honor.code"
-                    tagName="p"
-                    defaultMessage="By creating an account, you agree to the {tosAndHonorCode} and you acknowledge that {platformName} and each
-                    Member process your personal data in accordance with the {privacyPolicy}."
-                    description="Text that appears on registration form stating honor code and privacy policy"
-                    values={{
-                      platformName: this.state.platformName,
-                      tosAndHonorCode: (
-                        <Hyperlink destination={getConfig().TOS_AND_HONOR_CODE} target="_blank">
-                          {intl.formatMessage(messages['terms.of.service.and.honor.code'])}
-                        </Hyperlink>
-                      ),
-                      privacyPolicy: (
-                        <Hyperlink destination={getConfig().PRIVACY_POLICY} target="_blank">
-                          {intl.formatMessage(messages['privacy.policy'])}
-                        </Hyperlink>
-                      ),
-                    }}
-                  />
+              </Form.Group>
+            ) : null}
+            { this.state.showOptionalField ? this.getOptionalFields() : null }
+            <StatefulButton
+              type="submit"
+              variant="brand"
+              className="register-button-width mt-4"
+              state={submitState}
+              labels={{
+                default: intl.formatMessage(messages['create.account.button']),
+                pending: '',
+              }}
+              icons={{ pending: <FontAwesomeIcon icon={faSpinner} spin /> }}
```

Comment:
Do we need to add some sr-only text here since we have removed the text for the pending state? And can we use Paragon spinner instead? https://edx.github.io/paragon/components/spinner/

Reply:
I will add text and replace it with paragon spinner